python/cudf/cudf/_fuzz_testing/utils.py

```diff
@@ -1,16 +1,42 @@
 # Copyright (c) 2020, NVIDIA CORPORATION.
 import random
 
+import fastavro
+import numpy as np
 import pandas as pd
 
 import cudf
 from cudf.tests.utils import assert_eq
-from cudf.utils.dtypes import pyarrow_dtypes_to_pandas_dtypes
+from cudf.utils.dtypes import (
+    pandas_dtypes_to_cudf_dtypes,
+    pyarrow_dtypes_to_pandas_dtypes,
+)
 
 ALL_POSSIBLE_VALUES = "ALL_POSSIBLE_VALUES"
 
-
-def _generate_rand_meta(obj, dtypes_list):
+_PANDAS_TO_AVRO_SCHEMA_MAP = {
+    np.dtype("int8"): "int",
+    pd.Int8Dtype(): ["int", "null"],
+    pd.Int16Dtype(): ["int", "null"],
+    pd.Int32Dtype(): ["int", "null"],
+    pd.Int64Dtype(): ["long", "null"],
+    pd.BooleanDtype(): ["boolean", "null"],
+    pd.StringDtype(): ["string", "null"],
+    np.dtype("bool_"): "boolean",
+    np.dtype("int16"): "int",
+    np.dtype("int32"): "int",
+    np.dtype("int64"): "long",
+    np.dtype("O"): "string",
+    np.dtype("str"): "string",
+    np.dtype("float32"): "float",
+    np.dtype("float64"): "double",
+    np.dtype("<M8[ns]"): {"type": "long", "logicalType": "timestamp-millis"},
+    np.dtype("<M8[ms]"): {"type": "long", "logicalType": "timestamp-millis"},
+    np.dtype("<M8[us]"): {"type": "long", "logicalType": "timestamp-micros"},
```

Comment:
Why don't we have `<M8[s]>` ? Is it an AVRO limitation?

Reply:
yep, only milli and micro precisions are supported: https://avro.apache.org/docs/current/spec.html#Time+%28millisecond+precision%29